code/modules/mob/living/carbon/life.dm

```diff
@@ -245,6 +278,37 @@
 			if(!.)
 				return FALSE //to differentiate between no internals and active, but empty internals
 
+// Make corpses rot, emitting miasma
+/mob/living/carbon/proc/rot()
+	// Properly stored corpses shouldn't create miasma
+	if(istype(loc, /obj/structure/closet/crate/coffin)|| istype(loc, /obj/structure/closet/body_bag) || istype(loc, /obj/structure/bodycontainer))
+		return
+
+	// No decay if formaldehyde in corpse
+	if(reagents.has_reagent("formaldehyde", 20))

```

Comment:
I suggest 15u just so it takes one full syringe per body and 2 bodies per 30u bottle